.github/workflows/quality-control.yml

```
@@ -23,22 +25,40 @@ jobs:
       - name: Validate Composer
         run: composer validate
 
+  rector:
+    name: Run Rector
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.3
+          coverage: none
+
+      - name: Install composer dependencies
+        uses: ramsey/composer-install@v3
+
+      - name: Run Rector
+        run: vendor/bin/rector process --no-ansi --dry-run --no-diffs --no-progress-bar
```

no `--dry-run` this will get it committed alongside the style changes in the next step
```suggestion
        run: vendor/bin/rector process --no-ansi --no-progress-bar
```